x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigInitialNode.java

```diff
@@ -0,0 +1,352 @@
+            try (SecureString transportKeystorePassword = keystorePassword()) {
+                KeyStore transportKeystore = KeyStore.getInstance("PKCS12");
+                transportKeystore.load(null);
+                // the PKCS12 keystore and the contained private key use the same password
+                transportKeystore.setKeyEntry(TRANSPORT_AUTOGENERATED_KEYSTORE_NAME, transportKeyPair.getPrivate(),
+                        transportKeystorePassword.getChars(), new Certificate[]{transportCert});
+                fullyWriteFile(transportKeystoreOutput, stream -> transportKeystore.store(stream, transportKeystorePassword.getChars()));
+                // no danger of overwrite because Security was not initially enabled, and this secure setting can only exist in that case
+                nodeKeystore.setString("xpack.security.transport.ssl.keystore.secure_password", transportKeystorePassword.getChars());
+            } finally {
+                nodeKeystore.save(env.configFile(), nodeKeystorePassword.getChars());
+            }
+            Path httpKeystoreOutput = instantAutoConfigDir.resolve(HTTP_AUTOGENERATED_KEYSTORE_NAME + ".p12");
+            try (SecureString httpKeystorePassword = keystorePassword()) {
+                KeyStore httpKeystore = KeyStore.getInstance("PKCS12");
+                httpKeystore.load(null);
+                // the keystore contains both the node's and the CA's private keys
+                // both keys are encrypted using the same password as the PKCS12 keystore they're contained in
+                httpKeystore.setKeyEntry(HTTP_AUTOGENERATED_KEYSTORE_NAME + "_ca", httpCAKeyPair.getPrivate(),
+                        httpKeystorePassword.getChars(), new Certificate[]{httpCACert});
+                httpKeystore.setKeyEntry(HTTP_AUTOGENERATED_KEYSTORE_NAME, httpKeyPair.getPrivate(),
+                        httpKeystorePassword.getChars(), new Certificate[]{httpCert, httpCACert});
+                fullyWriteFile(httpKeystoreOutput, stream -> httpKeystore.store(stream, httpKeystorePassword.getChars()));
+                // no danger of overwrite because Security was not initially enabled, and this secure setting can only exist in that case
+                nodeKeystore.setString("xpack.security.http.ssl.keystore.secure_password", httpKeystorePassword.getChars());
+            } finally {
+                nodeKeystore.save(env.configFile(), nodeKeystorePassword.getChars());
+            }
+        }
+
+        {
+            Path transportTruststoreOutput = instantAutoConfigDir.resolve(TRANSPORT_AUTOGENERATED_TRUSTSTORE_NAME + ".p12");
+            final KeyStore transportTruststore = KeyStore.getInstance("PKCS12");
+            transportTruststore.load(null);
+            transportTruststore.setCertificateEntry(TRANSPORT_AUTOGENERATED_TRUSTSTORE_NAME, transportCert);
+            fullyWriteFile(transportTruststoreOutput, stream -> transportTruststore.store(stream, new char[0]));
```

Comment:
fullyWriteFile writes to disk with permissions set to 600. For packages, we should allow group (`elasticsearch`) to also read/write ( as we do with other files in the config dir ). We could figure out a way to branch but I think it is not that bad an idea to store these as `660` eitherway.